src/dmd/link.d

```diff
@@ -337,11 +348,38 @@ public int runLINK()
             else
                 cmdbuf.writestring("nul");
             cmdbuf.writeByte(',');
-            for (size_t i = 0; i < global.params.libfiles.dim; i++)
+
+            static bool isLibrarySearchPath(const(char)* option)
             {
-                if (i)
-                    cmdbuf.writeByte('+');
-                writeFilename(&cmdbuf, global.params.libfiles[i]);
+                return option[0] == '+' && option[strlen(option) - 1] == '\\';
+            }
+            {
+                bool first = true;
```

Comment:
I think it would be better to move this code into a separate function to avoid adding even more code to this large function. It would also get rid of the slightly irritating indentation.